core/watcher/eventqueue/uuid.go

```diff
@@ -0,0 +1,133 @@
+// Copyright 2023 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package eventqueue
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+
+	"github.com/juju/collections/transform"
+	"github.com/juju/errors"
+	"gopkg.in/tomb.v2"
+
+	"github.com/juju/juju/core/changestream"
+)
+
+// UUIDWatcher watches for changes to a database table.
+// Any time rows change in the watched table, the changed UUIDs are emitted.
+type UUIDWatcher struct {
+	*BaseWatcher
+
+	out       chan []string
+	tableName string
+	selectAll string
+}
+
+// NewUUIDWatcher returns a new watcher that receives changes from the
+// input base watcher's db/queue when rows in the input table change.
+func NewUUIDWatcher(base *BaseWatcher, tableName string) *UUIDWatcher {
+	w := &UUIDWatcher{
+		BaseWatcher: base,
+		out:         make(chan []string),
+		tableName:   tableName,
+		selectAll:   fmt.Sprintf("SELECT uuid FROM %s", tableName),
+	}
+
+	w.tomb.Go(w.loop)
+	return w
+}
+
+// Changes returns the channel on which the UUIDs for
+// changed rows are sent to downstream consumers.
+func (w *UUIDWatcher) Changes() <-chan []string {
+	return w.out
+}
+
+func (w *UUIDWatcher) loop() error {
+	subscription, err := w.eventQueue.Subscribe(
+		changestream.Namespace(w.tableName, changestream.Create|changestream.Update|changestream.Delete))
```

Comment:
I wonder if we could have `changestream.AllTypes` (or similar) instead of `changestream.Create|changestream.Update|changestream.Delete`

Reply:
Done.